tests/test_compat.py

```diff
@@ -0,0 +1,76 @@
+import pytest
+import pyhf
+import pyhf.compat
+import pyhf.readxml
+
+
+def test_interpretation():
+    interp = pyhf.compat.interpret_rootname('gamma_foo_0')
+    assert interp['constrained'] == 'n/a'
+    assert not interp['is_scalar']
+    assert interp['name'] == 'foo'
+    assert interp['element'] == 0
+
+    interp = pyhf.compat.interpret_rootname('alpha_foo')
+    assert interp['constrained']
+    assert interp['is_scalar']
+    assert interp['name'] == 'foo'
+    assert interp['element'] == 'n/a'
+
+    interp = pyhf.compat.interpret_rootname('mu')
+    assert not interp['constrained']
+    assert interp['is_scalar']
+    assert interp['name'] == 'mu'
+    assert interp['element'] == 'n/a'
+
+    interp = pyhf.compat.interpret_rootname('Lumi')
+    assert interp['name'] == 'lumi'
+
+    interp = pyhf.compat.interpret_rootname('Lumi')
+    assert interp['name'] == 'lumi'
+
+    with pytest.raises(ValueError):
+        pyhf.compat.interpret_rootname('gamma_foo')
+
+    with pytest.raises(ValueError):
+        pyhf.compat.interpret_rootname('alpha_')
+
+
+def test_torootname():
+    m1 = pyhf.simplemodels.correlated_background([5], [50], [52], [48])
+    m2 = pyhf.simplemodels.hepdata_like([5], [50], [7])
+    m3 = pyhf.simplemodels.hepdata_like([5, 6], [50, 50], [7, 8])
```

Comment:
It would be nice if we could get PR #1438 in before this so that these `hepdata_like` calls could just be renamed appropriately.